src/zenml/cli/utils.py

```diff
@@ -2748,3 +2748,19 @@ def is_jupyter_installed() -> bool:
         return True
     except pkg_resources.DistributionNotFound:
         return False
+
+
+def requires_mac_env_var_warning() -> bool:
+    """Checks if a warning needs to be shown for a local Mac server.
+
+    This is for the case where a user is on a MacOS system, trying to run a
+    local server but is missing the `OBJC_DISABLE_INITIALIZE_FORK_SAFETY`
+    environment variable.
+
+    Returns:
+        bool: True if a warning needs to be shown, False otherwise.
+    """
+    return (
+        not os.getenv("OBJC_DISABLE_INITIALIZE_FORK_SAFETY")
+        and sys.platform == "darwin"
```

Comment:
I think this env variable is only necessary on M1 macs, can we detect that somehow?

Reply:
Actually this is wrong, but it depends on the system version which we could detect I assume?

Reply:
What do you mean system version? like the operating system version? Sonoma vs snow leopard etc? We can detect that with:

```python
import platform

# Get the macOS version
macos_version = platform.mac_ver()[0]
```

I didn't realise that this was only a thing above a certain version. I'll look into when that was introduced.

Reply:
Yes, the link you mention in the docs explains that this is relevant for macos 10.13+ only